src/main/java/core/basesyntax/impl/WritingToNewFileImpl.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.impl;
+
+import core.basesyntax.service.WritingToNewFile;
+import core.basesyntax.db.FruitShopDao;
+import core.basesyntax.service.CantWorkWithThisFileException;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class WritingToNewFileImpl implements WritingToNewFile {
+    private FruitShopDao fruitShopDao;
+
+    public WritingToNewFileImpl(FruitShopDao fruitShopDao) {
+        this.fruitShopDao = fruitShopDao;
+    }
+
+    @Override
+    public void writing(String fileName) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(fileName))) {
+            writer.write("fruit,quantity");
+            writer.newLine();
+            for (int i = 0; i < fruitShopDao.storageSize(); i++) {
```

Comment:
Directly accessing the storage size and data from the DAO is not a good encapsulation practice. Consider providing a method in the DAO that returns a list of entries or a map instead of exposing the internal data structure.